desc/objectives/linear_objectives.py

```diff
@@ -128,7 +134,17 @@ def build(self, eq, use_jit=True, verbose=1):
                         "yellow",
                     )
                 )
+
         self._dim_f = idx.size
+        if self.target is not None:  # rearrange given target to match modes order
+            if self._modes is True or self._modes is False:
+                raise RuntimeError(
```

Comment:
Making sure I understand why this error is necessary: we can't use the `target` values that the user passes in because they could be assuming a different order than the basis, right?

Reply:
basically yea, we wanna make sure that the target and modes arrays are corresponding to eachother, and so if a target is manually passed in you better be also passing in the modes array as a full array